### ISSUE TITLE
Error while running flow over docker infrastructure

### ISSUE THREAD

**Issue by davidesba** [PrefectHQ/prefect#14490]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

Any flow run using docker infrastructure fails at container start. At container start, if there is any default volumes configuration in the job template, every string will be changed  to boolean:

```
Docker config: command='prefect flow-run execute' env={'PREFECT_API_URL': 'http://prefect:4200/api', 'PREFECT__FLOW_RUN_ID': 'fe991e9b-768c-4eaf-813f-66c8c33d0908', 'IS_PREFECT_DOCKER': 'true', 'PREFECT_LOGGING_EXTRA_LOGGERS': 'iadmd'} labels={'io.prefect.flow-run-id': 'fe991e9b-768c-4eaf-813f-66c8c33d0908', 'io.prefect.flow-run-name': 'mindful-falcon', 'io.prefect.version': '3.0.0rc10', 'io.prefect.deployment-id': '09a6157a-67d9-4b42-90e4-d84efdecb818', 'io.prefect.deployment-name': 'iatsapp1:info:app-config', 'io.prefect.deployment-updated': '2024-07-04T11:13:40.503357Z', 'io.prefect.flow-id': 'e4765c6b-59ba-487b-8ccc-6dafc5eff5d3', 'io.prefect.flow-name': 'info:app-config'} name='mindful-falcon' image='registry-docker:5000/iadm:2.17.2' registry_credentials=None image_pull_policy='Always' networks=[] network_mode=None auto_remove=False volumes=[True, True, True, True] stream_output=True mem_limit=None memswap_limit=None privileged=False
```

### Reproduction

Create a docker worker with default volumes in the job template, and run any flow:

```
"volumes": {
    "type": "array",
    "items": {
      "type": "string"
    },
    "title": "Volumes",
    "default": [
     "/var:/var",
     "/etc:/etc",
     "/var/log/iadm:/var/log/iadm",
     "/mnt:/mnt"
    ],
    "example": [
     "/my/local/path:/path/in/container"
    ],
    "description": "A list of volume to mount into created containers."
  },
```
Same happens if instead of setting the volumes in the job template, the volumes are set in the deployment as job variables:

```python
from prefect import flow
from prefect.docker import DockerImage

@flow
def my_flow():
    print("Hello from a Docker container!")

if __name__ == "__main__":
    my_flow.deploy(
        name="docker-volume-example",
        work_pool_name="my-docker-pool",
        image=DockerImage(
            name="my_image",
            volumes=["/host/path:/container/path"]
        )
    )
```

### Error

```
Worker 'iatsapp1' submitting flow run '3b09f670-89ae-4679-aec2-bd841c30993c'
12:38:06 PM
prefect.flow_runs.worker
Failed to submit flow run '3b09f670-89ae-4679-aec2-bd841c30993c' to infrastructure.
Traceback (most recent call last):
  File "/opt/iadm/lib/python3.11/site-packages/prefect/workers/base.py", line 951, in _submit_run_and_capture_errors
    result = await self.run(
             ^^^^^^^^^^^^^^^
  File "/opt/iadm/lib/python3.11/site-packages/prefect_docker/worker.py", line 405, in run
    container, created_event = await run_sync_in_worker_thread(
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/iadm/lib/python3.11/site-packages/prefect/utilities/asyncutils.py", line 270, in run_sync_in_worker_thread
    result = await anyio.to_thread.run_sync(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/iadm/lib/python3.11/site-packages/anyio/to_thread.py", line 56, in run_sync
    return await get_async_backend().run_sync_in_worker_thread(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/iadm/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 2177, in run_sync_in_worker_thread
    return await future
           ^^^^^^^^^^^^
  File "/opt/iadm/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 859, in run
    result = context.run(func, *args)
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/iadm/lib/python3.11/site-packages/prefect/utilities/asyncutils.py", line 278, in call_with_mark
    return call()
           ^^^^^^
  File "/opt/iadm/lib/python3.11/site-packages/prefect_docker/worker.py", line 557, in _create_and_start_container
    raise exc
  File "/opt/iadm/lib/python3.11/site-packages/prefect_docker/worker.py", line 554, in _create_and_start_container
    container = self._create_container(docker_client, **container_settings)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/iadm/lib/python3.11/site-packages/prefect_docker/worker.py", line 712, in _create_container
    container = docker_client.containers.create(name=name, **kwargs)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/iadm/lib/python3.11/site-packages/docker/models/containers.py", line 916, in create
    create_kwargs = _create_container_args(kwargs)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/iadm/lib/python3.11/site-packages/docker/models/containers.py", line 1154, in _create_container_args
    create_kwargs['volumes'] = [
                               ^
  File "/opt/iadm/lib/python3.11/site-packages/docker/models/containers.py", line 1155, in <listcomp>
    _host_volume_from_bind(v) for v in volumes
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/iadm/lib/python3.11/site-packages/docker/models/containers.py", line 1161, in _host_volume_from_bind
    drive, rest = ntpath.splitdrive(bind)
                  ^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen ntpath>", line 169, in splitdrive
TypeError: expected str, bytes or os.PathLike object, not bool
12:38:22 PM
prefect.flow_runs.worker
Completed submission of flow run '3b09f670-89ae-4679-aec2-bd841c30993c'
12:38:22 PM
prefect.flow_runs.worker
Reported flow run '3b09f670-89ae-4679-aec2-bd841c30993c' as crashed: Flow run could not be submitted to infrastructure
```

### Versions (`prefect version` output)

```Text
Version:             3.0.0rc10
API version:         0.8.4
Python version:      3.11.0
Git commit:          200cf212
Built:               Wed, Jul 3, 2024 1:39 PM
OS/Arch:             linux/x86_64
Profile:             default
Server type:         server
Pydantic version:    2.7.3
```

**Comment by zzstoatzz** [comment]
thank you for the issue @davidesba! I have reproduced this and am looking into this now

---

note that in the second example, before I get such an error I get an error from docker
```
"/Users/nate/github.com/prefecthq/prefect/src/prefect/utilities/dockerutils.py", line 172, in build_image
    events = client.api.build(
             ^^^^^^^^^^^^^^^^^
TypeError: BuildApiMixin.build() got an unexpected keyword argument 'volumes'
```

but it's fine because I reproduce fine updating/overriding my work pool with the example volumes.

**Comment by zzstoatzz** [comment]
hi @davidesba - this should be fixed by #14516 and `prefect-docker==0.6.0rc3` has been released with this fix
